feat: added an article about basic proofing concepts

Shvandre opened the pull request:
## Description

Added an article about basic proofing concepts

Closes <--link to issue]-->.

## Checklist

- [ ] I have created an issue.
- [x] I am working on content that aligns with the [Style guide](https://docs.ton.org/v3/contribute/style-guide/).
- [x] I have reviewed and formatted the content according to [Content standardization](https://docs.ton.org/v3/contribute/content-standardization/).
- [x] I have reviewed and formatted the text in the article according to [Typography](https://docs.ton.org/v3/contribute/typography/).

novusnota commented on docusaurus.config.js:
```diff
@@ -50,6 +50,7 @@ const config = {
   baseUrl: '/',
```
Oh, neat

Shvandre replied:
Thank you🤗

novusnota reviewed (commented):
If we exclude some possible editing redundancies like `---` separators or wrong links to pages (should be relative urls as seen on the website, not local file paths), I'd suggest some structural changes if you don't mind.

First off, could you start by briefly describing "what is a proof"? Upon reading this article, I now know more about blocks and pruned cells than I did learn about proofs.

For example, you can begin the article in the following way:

```
# Basic proofing concepts

Proof is …. To proof something means to do X, Y, Z …. Everything on TON Blockchain is stored in [cells](/relative/doc/link), so you can prove almost anything, as cells are very proof-friendly.

:::info
Before proceeding, familiarize yourself with the following concepts:
- [Cells](/relative/doc/link)
- [Exotic cells](/relative/doc/link)
:::

This article covers situations when you want to verify the proof in a smart contract on-chain. However, the same techniques can be used for validating proofs off-chain.
```

Then, there should be a first h2 `##`, describing "several key points" and delving into more proofing details.

From there, I'd go directly into the contents of the "High-level overview", and describe the block structure as we go and as it's necessary.

Finally, I'd give concrete examples on proofing something. At least one example will do.

Just my 2 cents, feel free to refute and or suggest your structure :)

Shvandre commented:
@novusnota sorry for spelling mistake in commit (78c2575) message :( 
I am not sure if it is a good idea to force push to correct this mistake.
So
- Fixed links and removed separators `---`
- Described what is a proof
- Added some real-world example, not a full example, but the full one is too big for this article

As for describing block structure as we go, I think that the structure of the block should be described before, because not everything can be proven, and in my subjective opinion it is easier to proceed not from the desire to prove something, but from the possibility

reveloper commented:
@Shvandre, @aigerimu, could you please exclude unrelated changes from this PR?
<img width="356" height="83" alt="Screenshot 0007-08-06 at 13 49 46" src="https://github.com/user-attachments/assets/17af1b85-9581-4109-89ec-1814d0758f3e" />

aigerimu commented:
Thank you!
